clients/filesystem-hadoop3/src/main/java/org/apache/gravitino/filesystem/hadoop/GravitinoVirtualFileSystem.java

```diff
@@ -294,7 +296,27 @@ private Path getActualPathByIdentifier(
 
         return new Path(storageLocation);
       } else {
-        return new Path(virtualPath.replaceFirst(virtualLocation, storageLocation));
+        // if the storage location ends with "/",
+        // we should handle the conversion specially
+        if (storageLocation.endsWith(SLASH)) {
+          String subPath = virtualPath.substring(virtualLocation.length());
+          // For example, if the virtual path is `gvfs://fileset/catalog/schema/test_fileset/ttt`,
+          // and the storage location is `hdfs://cluster:8020/user/`,
+          // we should replace `gvfs://fileset/catalog/schema/test_fileset` with
+          // `hdfs://localhost:8020/`.
```

Comment:
Fixed. It's 'hdfs://localhost:8020/user' without tailing slash, because the virtual path `gvfs://fileset/catalog/schema/test_fileset/ttt` already has a slash before the `ttt`.